src/Components/test/Microsoft.AspNetCore.Components.Test/ComponentBaseTest.cs

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.Test
```

Comment:
I've added some tests for that, but they are pretty limited right now. They would still be passing even if we *were* reporting the cancellation as exceptions, since the tests don't have any way to observe async exceptions from lifecycle methods.

The underlying limitation is that we don't have proper error reporting for that. I've filed a separate issue to track dealing with this: https://github.com/aspnet/AspNetCore/issues/4964. When that gets implemented, the unit tests I've added in this PR will only continue to pass if the task cancellation handling logic is correct (which I think it is anyway).

Reply:
Yeah I think that's good enough for now.